src/components/Checkout/PayPalExpress.vue

```diff
@@ -16,58 +18,119 @@ export default {
+			paypalScript.src = 'https://www.paypalobjects.com/api/checkout.js';
+		}
+		return {
+			script: [
+				paypalScript
+			]
+		};
 	},
 	mounted() {
-		console.log('paypal express component mounted');
-		// this.initializePaypal();
+		this.initializePaypal();
+	},
+	watch: {
+		amount() {
+			this.initializePaypal();
+		}
 	},
 	methods: {
 		initializePaypal() {
 			// ensure paypal is loaded before calling
-			// Server render is fine
-			// Init from Mounted Hook when navigating from Ui Page fails as paypal checkout.js is still loading...
-			const vm = this;
-			console.log('mounted');
-			console.log(window.paypal);
-			if (typeof paypal !== 'undefined') {
-				paypal.Button.render(
-					{
-						env: 'sandbox',
-						commit: true,
-						payment: () => {
-							console.log('payment stage');
-							return new paypal.Promise((resolve, reject) => {
-								vm.apollo.query({
-									query: getPaymentToken
-								}).then(({ data }) => {
-									if (data) {
-										console.log(data);
-										if (data.errors) {
-											reject(data);
-										}
-										resolve(data.shop.getPaymentToken);
+			this.ensurePaypal = window.setInterval(() => {
+				if (typeof paypal !== 'undefined') {
+					this.renderPaypalButton();
+				}
+			}, 200);
+		},
+		renderPaypalButton() {
+			// clear ensurePaypal interval
+			window.clearInterval(this.ensurePaypal);
+			// render paypal button
+			paypal.Button.render(
+				{
+					// TODO: Wire up switch for Prod
+					env: 'sandbox',
+					commit: true,
+					payment: () => {
+						console.log('payment stage');
+						return new paypal.Promise((resolve, reject) => {
+							// Use updated vars on render
+							this.apollo.query({
+								query: getPaymentToken,
+								variables: {
+									amount: numeral(this.amount).format('0.00'),
+								}
+							}).then(({ data }) => {
+								if (data) {
+									console.log(data);
+									if (data.errors) {
+										reject(data);
 									}
-								});
+									resolve(data.shop.getPaymentToken);
+								}
 							});
-						},
-						onAuthorize: data => {
-							console.log('authorized stage');
-							console.log(data);
-						},
-						style: {
-							color: 'blue',
-							shape: 'rect',
-							size: 'large'
-						}
+						});
+					},
+					onAuthorize: data => {
+						console.log('authorized stage');
+						console.log(data);
+
+						return new paypal.Promise((resolve, reject) => {
+							this.apollo.mutate({
+								mutation: depositAndCheckout,
+								variables: {
+									amount: numeral(this.amount).format('0.00'),
+									token: data.paymentToken,
+									payerId: data.payerID
+								},
+							})
+								.then(ppResponse => {
+									console.log(ppResponse);
+									if (ppResponse.errors) {
+										// Transaction is complete
+										// Currently returns error from findAccountForIpn
+										// "Can not process without account id"
+										console.log(ppResponse);
```

Comment:
@jsaur I was getting an error as mentioned in the comments on the VM. Hopefully it's just that context that fails...so I'm looking to test this out in Dev

Reply:
Sounds reasonable, let's see what happens. 👍